test-case/latency-metrics.sh

```diff
@@ -0,0 +1,305 @@
+    skip_test "No playback or capture JACK port is specified."
+  fi
+}
+
+set_alsa()
+{
+  reset_sof_volume
+
+  # If MODEL is defined, set proper gain for the platform
+  if [ -z "$MODEL" ]; then
+    dlogw "NO MODEL is defined. Please define MODEL to run alsa_settings/MODEL.sh"
+  else
+    set_alsa_settings "$MODEL"
+  fi
+}
+
+# set/update commands in case the test iterates or sweep over a range
+compose_commands()
+{
+  TIMEOUT_RUN=(timeout "--kill-after" "${RUN_PERIOD_MAX}")
+  [ -n "${run_verbose}" ] && TIMEOUT_RUN=("${TIMEOUT_RUN[@]}" "${run_verbose}")
+
+  [ -n "${run_verbose}" ] && JACKD_OPTIONS=("${JACKD_OPTIONS[@]}" "${run_verbose}")
+  JACKD_RUN=(jackd "${JACKD_OPTIONS[@]}" -d "${JACKD_BACKEND}" "${JACKD_BACKEND_OPTIONS[@]}" -p "${jackd_period}")
+
+  TIMEOUT_JACKD_RUN=("${TIMEOUT_RUN[@]}" "${JACKD_TIMEOUT}" "${JACKD_RUN[@]}")
+}
+
+check_jackd_configured()
+{
+  dlogi "Check JACK audio server can be started"
+
+  compose_commands
+
+  dlogc "${TIMEOUT_JACKD_RUN[@]}"
+  "${TIMEOUT_JACKD_RUN[@]}" & jackdPID=$!
+
+  sleep ${WAIT_JACKD}
+
+  dlogc "jack_lsp"
+  jack_lsp -AclLpt
+
+  dlogi "Waiting jackd to stop without a client"
+  wait ${jackdPID}
+}
+
+collect_latency_data()
+{
+  dlogi "Start collecting latency data"
+
+  compose_commands
+
+  dlogc "${TIMEOUT_JACKD_RUN[@]}"
+  "${TIMEOUT_JACKD_RUN[@]}" 2>&1 | \
+      tee >(AWKPATH="${TESTLIB}:${AWKPATH}" \
+          gawk -f "${TESTLIB}/jackd_events.awk" > "${EVENTS_JSON}") & jackdPID="$!"
+
+  sleep ${WAIT_JACKD}
+
+  dlogc "jack_iodelay"
+  "${TIMEOUT_RUN[@]}" "${IODELAY_TIMEOUT}" \
+      stdbuf -oL -eL jack_iodelay 2>&1 | \
+      tee >(AWKPATH="${TESTLIB}:${AWKPATH}" \
+          gawk -f "${TESTLIB}/jack_iodelay_metrics.awk" > "${METRICS_JSON}") & iodelayPID="$!"
+
+  sleep ${WAIT_IODELAY}
+  if [ "${set_loopback}" == 1 ]; then
+      dlogi "jack_connect: [JACK loopback]<==>[jack_delay]"
+      jack_connect jack_delay:out jack_delay:in
+  else
+      dlogi "jack_connect: [${port_capture}]==>[jack_delay]==>[${port_playback}]"
+      jack_connect jack_delay:out "${port_playback}" && jack_connect jack_delay:in "${port_capture}"
+  fi
+
+  dlogi "Latency data collection"
+  wait ${jackdPID} ${iodelayPID}
+
+  dlogi "Latency data collection completed for period=${jackd_period} frames."
+}
+
+report_start()
+{
+  dlogi "Compose ${RESULT_JSON}"
+  printf '{"options":{%s}, "metrics":[' "$(options2json)" > "${RESULT_JSON}"
+}
+
+json_next_sep=""
+
+report_metric()
+{
+  printf '%s{"period_frames":%d, "periods":%d, "rate":%d, ' \
+    "${json_next_sep}" "${jackd_period}" "${jackd_nperiods}" "${rate}" >> "${RESULT_JSON}"
+
+  if [ ! -f "${METRICS_JSON}" ] || [ "$(grep -ce 'metric_name' "${METRICS_JSON}")" -lt 1 ]; then
+    printf '"probes":0, "xruns":0}' >> "${RESULT_JSON}"
+    [ -f "${METRICS_JSON}" ] && rm "${METRICS_JSON}"
+    [ -f "${EVENTS_JSON}" ] && rm "${EVENTS_JSON}"
+    [ "${trial_mode}" -eq 0 ] && skip_test "No latency metrics collected"
```

Comment:
Nit:
```suggestion
    [ "${trial_mode}" -ne 0 ] || skip_test "No latency metrics collected"
```
See above and #312